main.tf

```diff
@@ -13,8 +13,8 @@ resource "aws_security_group" "database" {
   vpc_id = var.vpc_id
 
   ingress {
-    from_port       = 5432
-    to_port         = 5432
+    from_port       = var.engine == "mysql" ? 3306 : 5432
```

Comment:
I'd rather make the port explicitly a variable with a default that matches the default engine. That will support the possibility of custom ports. It _might_ even be possible to default one based on the other?

Reply:
I do like that better. I originally tried doing that with:

```terraform
variable "port" {
  default = var.engine == "mysql" ? 3306 : 5432
}
```

but I got an error saying I can't use variables there. But we could make it default to `-1` or something, falling back to engine-specific defaults in `main.tf`.

Reply:
Updated to allow custom ports, falling back to engine defaults.